server/workflowarchive/archived_workflow_server.go

```diff
@@ -70,7 +69,7 @@ func (w *archivedWorkflowServer) ListArchivedWorkflows(ctx context.Context, req
 				break
 			}
 			// TODO second pass filtering?
-			allowed, err := authorizer.CanI("get", workflow.WorkflowPlural, wf.Namespace, wf.Name)
+			allowed, err := auth.CanI(ctx, "get", workflow.WorkflowPlural, wf.Namespace, wf.Name)
```

Comment:
I understand this change, but it will make N calls to the K8S API under the hood. This will make the request very slow or not work at oll.

Reply:
Is there a way to batch multiple resources into a single `CanI` call? If not, what do you think the right approach would be? The existing implementation has the same N calls problem (as far as I can tell).

Reply:
Would it be wrong to do something like this?
```suggestion
			allowed, err := auth.CanI(ctx, "get", workflow.WorkflowPlural, wf.Namespace)
```
And only make the call once per namespace?

Reply:
not sure, I'm not an RBAC expert, I'd like to hear from @jessesuen on this